openedx/core/djangoapps/user_authn/views/tests/test_register.py

```diff
@@ -80,51 +80,7 @@ def setUp(self):  # pylint: disable=arguments-differ
         super(RegistrationViewValidationErrorTest, self).setUp()
         self.url = reverse("user_api_registration")
 
-    @mock.patch('openedx.core.djangoapps.user_authn.views.register.check_account_exists')
-    def test_register_retired_email_validation_error(self, dummy_check_account_exists):
-        dummy_check_account_exists.return_value = []
-        # Register the first user
-        response = self.client.post(self.url, {
-            "email": self.EMAIL,
-            "name": self.NAME,
-            "username": self.USERNAME,
-            "password": self.PASSWORD,
-            "honor_code": "true",
-        })
-        self.assertHttpOK(response)
-
-        # Initiate retirement for the above user:
-        fake_requested_retirement(User.objects.get(username=self.USERNAME))
-
-        # Try to create a second user with the same email address as the retired user
-        response = self.client.post(self.url, {
-            "email": self.EMAIL,
-            "name": "Someone Else",
-            "username": "someone_else",
-            "password": self.PASSWORD,
-            "honor_code": "true",
-        })
-        self.assertEqual(response.status_code, 400)
-        response_json = json.loads(response.content.decode('utf-8'))
-        self.assertDictEqual(
-            response_json,
-            {
-                "email": [{
-                    "user_message": (
-                        u"It looks like {} belongs to an existing account. "
-                        "Try again with a different email address."
-                    ).format(
-                        self.EMAIL
-                    )
-                }]
-            }
-        )
-
-    def test_register_retired_email_validation_error_no_bypass_check_account_exists(self):
-        """
-        This test is the same as above, except it doesn't bypass check_account_exists.  Not bypassing this function
-        results in the same error message, but a 409 status code rather than 400.
```

Comment:
Can you explain why we had this difference before and why we don't need it?  Was the 400 case just never used for real Production code, or are you changing the result for certain cases?

Reply:
The change of status codes was already made as part of a previous refactor.  Once again, we are consolidating on updating our tests to no longer use the test-only `create_account` function and using the public registration API instead.